oauth/index.php

```diff
@@ -1,72 +1,91 @@
 <?php
 session_start();
-?>
+/**
+ * @author Denis CLAVIER <clavierd at gmail dot com>
+ * A modified verion by dimst23
+ */
 
-<!DOCTYPE html>
-<html>
-	<head>
-	    <link rel="stylesheet" type="text/css" href="./style.css">
-		<title>LDAP Connection Interface</title>
-	</head>
 
-	<body>
-		<center>
-			<table background="images/login.png" border="0" width="733" height="348" cellspacing="1" cellpadding="4">
-				<tr>
-					<td width="40%">&nbsp;</td>
-					
-					<td width="60%">
-						<table border="0" width="100%">
+// include our LDAP object
+require_once __DIR__.'/LDAP/LDAP.php';
+require_once __DIR__.'/LDAP/config_ldap.php';
 
-							<tr>
-								<td align="center">
-									<div class="LoginTitle">LDAP Authentification</div>
-						
+$prompt_template = new DOMDocument();
+$prompt_template->loadHTMLFile('form_prompt.html');
 
-								<form method="post" action="connexion.php">
-								
-										<table border="0" width="90%" cellpadding="1">
-												<tr>
-													<td colspan="2" align="left">
-														
-															<div class="messageLogin" align="center">
-																
-															</div>												 
-														&nbsp;
-													</td>
-												</tr>
-												<tr>
-													<td align="left" width="40%" class="LoginUsername">
-														Username:&nbsp;
-													</td>
-													<td width="60%">
-														<input type="text" name="user" size="25" value="" >
-													</td>
-												</tr>
-												<tr>
-													<td align="left" width="40%" class="LoginUsername">
-														Password:&nbsp;
-													</td>
-													<td width="60%">
-														<input type="password" name="password" size="25" value="" >
-													</td>
-												</tr>
-												<tr><td colspan="2">&nbsp;</td></tr>
-												<tr>
-													<td colspan="2" align="center"> <input type="submit" class="GreenButton" name="login" value="       Connect       " > </td>
-												</tr>
-												
-										    
-										</table>
-									</form>
-									
-							</td>
-							</tr>
-						</table>
-					
-					</td>
-				</tr>
-			</table>
-		</center>
-	</body>
-</html>
\ No newline at end of file
+
+function messageShow($html_template, $message = 'No Msg') {
+	$modification_node = $html_template->getElementsByTagName('div')->item(5);
+	$page_fragment = $html_template->createDocumentFragment();
+	$page_fragment->appendXML($message);
+
+	$modification_node->appendChild($page_fragment);
+
+	echo $html_template->saveHTML();
+}
+
+
+// Verify all fields have been filled 
+if (empty($_POST['user']) || empty($_POST['password'])) 
+{
+	if (empty($_POST['user'])) {
+		messageShow($prompt_template, 'Username field can\'t be empty.');
+	} else {
+		messageShow($prompt_template, 'Password field can\'t be empty.');
+	}
+}
+else
+{
+	// Check received data length (to prevent code injection) 
+	if (strlen($_POST['user']) > 15)
```

Comment:
The max username length should be increased to 64 . Because we have Windows AD user accounts "prename.name" in our directory.